src/dbnode/storage/shard.go

```diff
@@ -2270,18 +2283,20 @@ func (s *dbShard) Snapshot(
 
 func (s *dbShard) FlushState(blockStart time.Time) (fileOpState, error) {
 	s.flushState.RLock()
-	defer s.flushState.RUnlock()
 	if !s.flushState.bootstrapped {
+		s.flushState.RUnlock()
```

Comment:
Is this a hot enough path that this is necessary? Otherwise with multiple paths defer may be safer here

Reply:
Will update so only one RUnlock()